tests/components/mqtt/test_sensor.py

```diff
@@ -654,6 +654,67 @@ async def test_discovery_update_availability(hass, mqtt_mock_entry_no_yaml_confi
     )
 
 
+async def test_availability_with_shared_state_topic(
```

Comment:
We should move this test to `test_mixins.py` if we don't want to duplicate it for each platform.